src/gpu_fdinfo.cpp

```diff
@@ -168,12 +168,15 @@ void GPU_fdinfo::get_load() {
         static uint64_t previous_gpu_time, previous_time, now, gpu_time_now;
         static float power_usage_now, previous_power_usage;
 
+        now = os_time_get_nano();
+
+        if (now - previous_time < METRICS_UPDATE_PERIOD_MS * 1'000'000)
```

Comment:
I believe you're right, we don't need to check this often, but I'm not sure this is sufficient.
We might be consuming a full thread like this and how it was before?

Reply:
In both cases thread is using 100% of CPU core. The only way to fix it is to sleep like this in `nvidia.cpp`
https://github.com/flightlessmango/MangoHud/blob/afaa7d1b7b9212e8938e70c2ac223789832c96c6/src/nvidia.cpp#L227

![image](https://github.com/user-attachments/assets/47744f84-05d2-47f7-a7f9-ceaadd786fe4)

Reply:
Do you want me to use sleep and if yes, should I use METRICS_POLLING_PERIOD_MS (25ms) or METRICS_UPDATE_PERIOD_MS (500ms)?

Reply:
500ms is probably fine for this

Reply:
added sleep